app/models/user.rb

```diff
@@ -98,6 +98,25 @@ def token
     JsonWebToken.encode(user_id: self.id, roles: [], email: self.email, verified: verified)
   end
 
+  def self.from_social(data)
+    Rails.logger.info "************ email is = #{data[:email]}"
+    user = User.where(email: data[:email]).first
+
+    path = '/profile'
+    unless user
+      Rails.logger.info "!!!!!! path is = #{path}"
+      user = User.new(first_name: data[:first_name],
```

Comment:
Can we carry this first attr down one line, for consistency and readability, like this:

```ruby
user = User.new(
  first_name: data[:first_name],
  last_name: data[:last_name],
  email: data[:email],
  zip: data[:zip],
  password: data[:password]
)
```

Reply:
done.